scripts/functional-tests.sh

```diff
@@ -124,7 +138,7 @@ cleanup() {
 }
 
 init_test_script
-trap 'cleanup "${?}"; rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
+trap 'cleanup "${?}"; test -z $TEST_DIR_SET_BY_CALLER && rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
```

Comment:
We should be using `"${TEST_DIR_SET_BY_CALLER}"` to avoid potential bugs when bash expands this variable.

```suggestion
trap 'cleanup "${?}"; test -z "${TEST_DIR_SET_BY_CALLER}" && rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
```

Reply:
I also thought so at first, but: Not really.

```
$ test -z && echo true
true
```

Reply:
That's right, I forgot to quote it.

Reply:
Well I guess it does not hurt :)